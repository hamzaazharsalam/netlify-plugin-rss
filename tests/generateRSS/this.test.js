const fs = require('fs');
const path = require('path');

// resolve all files
const PUBLISH_DIR = path.join(__dirname, 'publishDir');
const dirToScan = '/blog';
// debug
// console.log({ publishDirPath, manifest });

// actual test
const pluginCore = require('../../pluginCore.js');
test('generateRSS works', async () => {
  const rssFeed = await pluginCore.generateRSS({
    dirToScan,
    PUBLISH_DIR,
    authorName: 'swyx',
    site_url: 'https://swyx.io',
    feed_url: 'https://swyx.io/rss.xml',
    rssFaviconUrl: 'https://swyx.io/favicon.png',
    title: 'swyx RSS Feed',
    rssDescription: 'swyx.io RSS Feed',
    // docs: 'http://example.com/rss/docs.html'
    // managingEditor = authorName,
    // webMaster = authorName,
    // copyright = new Date().getFullYear() + ' ' + authorName,
    // language = 'en',
    categories: ['Technology', 'JavaScript', 'React', 'Svelte'],
    // pubDate = new Date().toUTCString(),
    // ttl = '60' // at most refresh every hour http://www.therssweblog.com/?guid=20070529130637

    // // options for extractMetadataFromFile
    contentSelector: '.singlePost',
    publishDateSelector: '.postDate'

    // debugMode: false, // to future readers - turn this true to make pluginCore log out more stuff
    // testMode: true
  });
  expect(rssFeed).toMatchSnapshot();
});
