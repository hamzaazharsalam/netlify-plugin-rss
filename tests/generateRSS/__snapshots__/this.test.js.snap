// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateRSS works 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><rss xmlns:dc=\\"http://purl.org/dc/elements/1.1/\\" xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\" xmlns:atom=\\"http://www.w3.org/2005/Atom\\" version=\\"2.0\\"><channel><title><![CDATA[swyx RSS Feed]]></title><description><![CDATA[swyx.io RSS Feed]]></description><link>https://swyx.io</link><image><url>https://swyx.io/favicon.png</url><title>swyx RSS Feed</title><link>https://swyx.io</link></image><generator>RSS for Node</generator><lastBuildDate>Sun, 15 Mar 2020 04:30:06 GMT</lastBuildDate><atom:link href=\\"https://swyx.io/rss.xml\\" rel=\\"self\\" type=\\"application/rss+xml\\"/><pubDate>Sun, 15 Mar 2020 04:30:06 GMT</pubDate><copyright><![CDATA[2020 swyx]]></copyright><language><![CDATA[en]]></language><managingEditor><![CDATA[swyx]]></managingEditor><webMaster><![CDATA[swyx]]></webMaster><docs>http://example.com/rss/docs.html</docs><ttl>60</ttl><category><![CDATA[Technology]]></category><category><![CDATA[JavaScript]]></category><category><![CDATA[React]]></category><category><![CDATA[Svelte]]></category><item><title><![CDATA[This is our First Post]]></title><description><![CDATA[or, Career Advice For The Working Web Dev

I have been thinking a lot about the thing we call \\"the stack\\", one of many vague concepts web developers use when describing themselves. People call themselves \\"frontend\\", \\"backend\\" and \\"full stack\\" but there's no real consensus on what any of those mean.

What is the stack?

Part of the problem is that the \\"stack\\" is enormous: it includes at a minimum HTML, CSS and JavaScript. But how deep do you go? There are many server-side languages, ther]]></description><link>https://swyx.io/../tests/generateRSS/publishDir/blogs</link><guid isPermaLink=\\"true\\">https://swyx.io/../tests/generateRSS/publishDir/blogs</guid><pubDate>Sun, 15 Mar 2020 03:56:20 GMT</pubDate><content:encoded><![CDATA[
                <div class=\\"postDate\\">1 January, 2020</div>
                <h1>This is our First Post</h1>
                <div>
                  some words for our first post
                </div>
              ]]></content:encoded></item><item><title><![CDATA[This is our Second Post]]></title><description><![CDATA[or, Career Advice For The Working Web Dev

I have been thinking a lot about the thing we call \\"the stack\\", one of many vague concepts web developers use when describing themselves. People call themselves \\"frontend\\", \\"backend\\" and \\"full stack\\" but there's no real consensus on what any of those mean.

What is the stack?

Part of the problem is that the \\"stack\\" is enormous: it includes at a minimum HTML, CSS and JavaScript. But how deep do you go? There are many server-side languages, ther]]></description><link>https://swyx.io/../../tests/generateRSS/publishDir/blogs</link><guid isPermaLink=\\"true\\">https://swyx.io/../../tests/generateRSS/publishDir/blogs</guid><pubDate>Sun, 15 Mar 2020 03:55:10 GMT</pubDate><content:encoded><![CDATA[
                <div class=\\"postDate\\">1 February, 2020</div>
                <h1>This is our Second Post</h1>
                <div>
                  some words for our second post
                </div>
              ]]></content:encoded></item><item><title><![CDATA[This is our Third Post]]></title><description><![CDATA[or, Career Advice For The Working Web Dev

I have been thinking a lot about the thing we call \\"the stack\\", one of many vague concepts web developers use when describing themselves. People call themselves \\"frontend\\", \\"backend\\" and \\"full stack\\" but there's no real consensus on what any of those mean.

What is the stack?

Part of the problem is that the \\"stack\\" is enormous: it includes at a minimum HTML, CSS and JavaScript. But how deep do you go? There are many server-side languages, ther]]></description><link>https://swyx.io/../../tests/generateRSS/publishDir/blogs</link><guid isPermaLink=\\"true\\">https://swyx.io/../../tests/generateRSS/publishDir/blogs</guid><pubDate>Sun, 15 Mar 2020 03:56:47 GMT</pubDate><content:encoded><![CDATA[
                <div class=\\"postDate\\">1 March, 2020</div>
                <h1>This is our Third Post</h1>
                <div>
                  some words for our 3rd post
                </div>
              ]]></content:encoded></item></channel></rss>"
`;
